### ISSUE TITLE
Create and render highlights with range information

### ISSUE THREAD

**Issue by aassoiants** [nostrdocs/colighter#34]
- [ ] When creating a new highlight, we need to serialize range/context information. The idea is to store this information under a tag in nostr events, such that we can retrieve them later
```json
{
    "created_at": 1687315880,
    "content": "Newton’s theory of gravity allowed scientists to predict the future movements of objects like planets",
    "tags": [
      [
        "r",
        "https://www.blinkist.com/en/books/a-brief-history-of-time-en"
      ],
      [
        "range",
        "<SERIALIZED-RANGE-INFO>"
      ]
    ],
    "kind": 9802,
    "id": "e33d486f5332f78953b3f0578d0def1bb777d53f13fa323dd8e8c96d2116a4a3",
    "pubkey": "7e5230957749abdff0c5290e3b6b780579a5a18bccab3ff9364648a603e35cd9"
}
```

- [ ] On receiving / loading a highlight event from nostr, we should be able to deserialize this range information, such that we markup (recreate) the highlight within canvas and context
![image](https://github.com/nostrdocs/colighter/assets/11217077/9600d4b9-98f2-4d48-8a66-5b317f156d09)

**Comment by Extheoisah** [comment]
Current progress:
- [x] shortcut done
- [x] Right click done
- [ ] Button pop up needs more clarity. Not sure the ux or flow of this @aassoiants

**Comment by okjodom** [comment]
![image](https://github.com/nostrdocs/colighter/assets/11217077/792fca8e-2485-4289-9ada-7b1d1743e0a7)
Splitting out this issue:

**Comment by okjodom** [comment]
closed by #119
